wicket-core/src/main/java/org/apache/wicket/ajax/res/js/wicket-ajax-jquery.js

```diff
@@ -350,6 +350,44 @@
 	 */
 	Wicket.Ajax.Call = Wicket.Class.create();
 
+	Wicket.Ajax.Call._suspended = 0;
+
+	Wicket.Ajax.Call.currentNotify = undefined;
+
+	Wicket.Ajax.Call.suspend = function () {
+		if (typeof (Wicket.Ajax.Call.currentNotify) != "function") {
+			Wicket.Log.error("Can't suspend: no evaluation in process");
+			return;
+		}
+		Wicket.Ajax.Call._suspended++;
+		var notify = Wicket.Ajax.Call.currentNotify;
+		var released = false;
+
+		return function () {
+			// release only once
+			if (released === false) {
+				released = true;
+				Wicket.Ajax.Call._suspended--;
+				if (Wicket.Ajax.Call._suspended === 0) {
+					notify();
+				}
+			}
+		}
+	};
+
+	Wicket.Ajax.Call.getFunctionsExecuterStatus = function() {
+		if (Wicket.Ajax.Call._suspended) {
+			// suspended
+			return FunctionsExecuter.ASYNC;
+		} else {
+			// execution finished, cleanup the last notify
+			Wicket.Ajax.Call.currentNotify = undefined;
```

Comment:
currentNotify has to be cleared in either case, there's no need to keep it around any longer. Either a suspension was aquired at this point or not.
Actually we have to move the _suspended counter into the scope of currentNotify, otherwise we run intro troubles with parallel Ajax requests.

Reply:
✔️ changed

Reply:
I think this conflicts with https://github.com/apache/wicket/pull/384#discussion_r331761728
the currentNotify should be called either when the call is not async, or when last suspend/lock is released.